### PR TITLE
fix(dart_frog_gen): static dir inside dynamic dir

### PR DESCRIPTION
**Pull request by renancaraujo**
## Status

READY

## Description
It was reported on #134 that the generator doesn't recognize routes under directories in the following relationship:

```
/routes/[dynamic_dir]/non_dynamic_dir
```

This changes the generator to verify for files inside non-dynamic directories that are inside dynamic directories.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/dart_frog_gen/lib/src/build_route_configuration.dart [hunk 1]
```diff
@@ -105,11 +105,12 @@ List<RouteFile> _getRouteFilesForDynamicDirectories(
   String prefix = '',
 }) {
   final files = <RouteFile>[];
+  final thisIsDynamic = directory.isDynamicRoute;
```
Comment (wolfenrain) [inline packages/dart_frog_gen/lib/src/build_route_configuration.dart:108]:
```suggestion
  final isDynamic = directory.isDynamicRoute;
```

Calling it this feels strange as it doesn't refer to the current class but to a value out of it

Comment (felangel) [reply packages/dart_frog_gen/lib/src/build_route_configuration.dart:108]: Agreed

File: packages/dart_frog_gen/test/src/build_route_configuration_test.dart [hunk 2]
```diff
@@ -299,5 +299,99 @@ void main() {
...
+      final itemDirectory = Directory(path.join(userDirectory.path, 'item'))
+        ..createSync();
+      File(path.join(itemDirectory.path, 'index.dart')).createSync();
+      final configuration = buildRouteConfiguration(directory);
+      expect(
+        configuration.directories.map((d) => d.toJson()).toList(),
+        equals(expected),
+      );
+    });
+    test('includes dynamic static dynamic nested directory routes', () {
```
Comment (felangel) [inline packages/dart_frog_gen/test/src/build_route_configuration_test.dart:349]:
```suggestion
    
    test('includes dynamic/static/dynamic nested directory routes', () {
```

File: packages/dart_frog_gen/test/src/build_route_configuration_test.dart [hunk 3]
```diff
@@ -299,5 +299,100 @@ void main() {
...
+              'path':
+                  '../test/.fixtures/static_dynamic_nested/routes/index.dart',
+              'route': '/routes'
+            },
+            {
+              'name': '.._test_.fixtures_static_dynamic_nested_routes'
+                  r'_$user_item_index',
+              'path': '../test/.fixtures/static_dynamic_nested/routes/'
+                  '[user]/item/index.dart',
+              'route': '/item/<user>'
```
Comment (felangel) [inline packages/dart_frog_gen/test/src/build_route_configuration_test.dart:322]: shouldn't this be `/<user>/item`?

**Review by felangel — CHANGES_REQUESTED** [review]
I don't think these changes result in the intended behavior. I'm looking into this now as well

**Comment by felangel** [comment]
I opened https://github.com/VeryGoodOpenSource/dart_frog/pull/138 to address this. lmk what you think

**Comment by felangel** [comment]
Closing since this should be fixed in #138

**Pull request closed without merging**